### DISCUSSION TITLE
converting db.Decimal as boolean

### DISCUSSION THREAD

**Discussion by ivolodym** [prisma/prisma#18651] [Q&A]
Hi! I have next problem: in our database (PostgreSQL) flags represented as numeric values but as booleans in application.
Here is an example of problematic code:
model ExampleModel {
  id        String  @id(map: "pk_some_model_34567865") @db.VarChar(15)
  someText  String  @db.VarChar(100)
  isSolved  Boolean @db.Decimal(1, 0)
}
P.S.: I can't change anything in DB because a lot of other apps depends on it. How can I apply some adapter which will transform values to "true/false" when fetching and to "0/1" when interacting with DB. Prisma version 3.8.1

**Comment by nurul3101** [comment]
Hello @ivolodym 👋 

I think Prisma Client Extension can be helpful in your use case. 

You can use the query extension ([reference](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions/query#modify-a-specific-operation-in-a-specific-model)) and convert the where clause, you can read the where argument and convert the respective boolean values to decimal before invoking the query.

And you can use result extension ([reference](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions/result#add-a-custom-field-to-query-results)) to return a custom boolean field depending on the decimal value.

**Reply by nurul3101** [reply]
Hello @ivolodym 👋 

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by ivolodym** [reply]
Hello! Unfortunately it looks like impossible. We reviewed all documentation and few days explored framework code and not found anything that looks like solution.
P.S. we reworked all business logic to replace all boolean variables by numeric in models and algorithms. Worst technical solution, but only the one which working

**Reply by nurul3101** [reply]
Thanks for getting back and letting us know.
I'll check and try to come up with a Client Extension for this use case.

**Reply by ivolodym** [reply]
Something like Hibernate/NHibernate tools for mapping will be very useful for many other developers

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
